hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/persistence/AbstractReconSqlDBTest.java

```diff
@@ -36,52 +38,51 @@
 import org.jooq.SQLDialect;
 import org.jooq.impl.DSL;
 import org.jooq.impl.DefaultConfiguration;
-import org.junit.Assert;
-import org.junit.Before;
-import org.junit.Rule;
+import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.BeforeEach;
-import org.junit.rules.TemporaryFolder;
 
 import com.google.inject.AbstractModule;
 import com.google.inject.Guice;
 import com.google.inject.Injector;
 import com.google.inject.Module;
 import com.google.inject.Provider;
+import org.junit.jupiter.api.io.TempDir;
 
 /**
  * Class that provides a Recon SQL DB with all the tables created, and APIs
  * to access the DAOs easily.
  */
 public class AbstractReconSqlDBTest {
 
-  @Rule
-  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+  @TempDir
+  protected Path temporaryFolder;
```

Comment:
> shall I make it private and introduce a getter method which lead the changes in all 14 sub-classes

Yes, please make it `private`.  It only affects 4 subclasses.

Reply:
I made it private, surprisingly the subclasses are creating their own temporary folders and hence were not dependent on folders been create in Abstract class. (Except TestReconWithDifferentSqlDBs which uses its super constructor but does not use the variable temporaryFolder)